go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -327,13 +328,31 @@ func (c *addressManager) nodePrimaryAddrChanged() (bool, error) {
 
 // updateOVNEncapIP updates encap IP to OVS when the node primary IP changed.
 func (c *addressManager) updateOVNEncapIPAndReconnect() {
-	cmd := []string{
+	checkCmd := []string{
+		"get",
+		"Open_vSwitch",
+		".",
+		"external_ids:ovn-encap-ip",
+	}
+	encapIP, stderr, err := util.RunOVSVsctl(checkCmd...)
+	if err != nil {
+		klog.Warningf("Unable to retrieve configured ovn-encap-ip from OVS: %v, %q", err, stderr)
+	} else {
+		encapIP = strings.TrimSuffix(encapIP, "\n")
```

Comment:
so the issue was that this was acting up on any update even if the changedIP was already configured? That's a bit weird:
```
Node primary address changed to 10.0.141.44. Updating OVN encap IP.
```
I don't get why if the IP .44 was already configured we were trying to do another update for it? Seems like it needs a fix in the higher up layers of "what we consider as a nodeIP update?"

Reply:
uhm... so `nodePrimaryAddrChanged` returns a wrong value :-/

Reply:
I agree we should investigate why that happens, but I think it is orthogonal to this change. We should check the OVS config before we do something so drastic as to restart ovn-controller.

Reply:
I filed https://issues.redhat.com/browse/OCPBUGS-15538 to track that issue.